### ISSUE TITLE
Build failure with Rust 1.21

### ISSUE THREAD

**Issue by eclipseo** [rust-lang/regex#409]
Works with 1.20, but Rust 1.2.1 brings new errors:

```
Running `/usr/bin/rustc --crate-name regex src/lib.rs --emit=dep-info,link -C opt-level=3 --test -C metadata=7f59e7b00711eb77 -C extra-filename=-7f59e7b00711eb77 --out-dir /builddir/build/BUILD/regex-0.2.2/target/release/deps -L dependency=/builddir/build/BUILD/regex-0.2.2/target/release/deps --extern quickcheck=/builddir/build/BUILD/regex-0.2.2/target/release/deps/libquickcheck-f540df7766419ffe.rlib --extern thread_local=/builddir/build/BUILD/regex-0.2.2/target/release/deps/libthread_local-601d1665f83a533c.rlib --extern memchr=/builddir/build/BUILD/regex-0.2.2/target/release/deps/libmemchr-960e0874235de3a4.rlib --extern rand=/builddir/build/BUILD/regex-0.2.2/target/release/deps/librand-a93677b31beba6ec.rlib --extern regex_syntax=/builddir/build/BUILD/regex-0.2.2/target/release/deps/libregex_syntax-84cd735e49a09121.rlib --extern utf8_ranges=/builddir/build/BUILD/regex-0.2.2/target/release/deps/libutf8_ranges-108996c80301bb7f.rlib --extern lazy_static=/builddir/build/BUILD/regex-0.2.2/target/release/deps/liblazy_static-914213a2ef2edf6b.rlib --extern aho_corasick=/builddir/build/BUILD/regex-0.2.2/target/release/deps/libaho_corasick-2bf634c39a177989.rlib -Copt-level=3 -Cdebuginfo=2 -Clink-arg=-Wl,-z,relro,-z,now`

error: variable does not need to be mutable
   --> src/backtrack.rs:103:13
    |
103 |         let mut cache = &mut cache.backtrack;
    |             ^^^^^^^^^
    |
note: lint level defined here
   --> src/lib.rs:491:24
    |
491 | #![cfg_attr(test, deny(warnings))]
    |                        ^^^^^^^^
    = note: #[deny(unused_mut)] implied by #[deny(warnings)]

error: variable does not need to be mutable
   --> src/dfa.rs:465:13
    |
465 |         let mut cache = &mut cache.dfa;
    |             ^^^^^^^^^

error: variable does not need to be mutable
   --> src/dfa.rs:498:13
    |
498 |         let mut cache = &mut cache.dfa_reverse;
    |             ^^^^^^^^^

error: variable does not need to be mutable
   --> src/dfa.rs:532:13
    |
532 |         let mut cache = &mut cache.dfa;
    |             ^^^^^^^^^

error: variable does not need to be mutable
   --> src/pikevm.rs:112:13
    |
112 |         let mut cache = &mut cache.pikevm;
    |             ^^^^^^^^^

error: variable does not need to be mutable
   --> src/pikevm.rs:344:25
    |
344 |                     let mut t = &mut nlist.caps(ip);
    |                         ^^^^^

error: aborting due to 6 previous errors

error: Could not compile `regex`.
```

**Comment by eclipseo** [comment]
My bad, it's already been fixed in master, just not released yet. Closing.

**Comment by BurntSushi** [comment]
This is fixed on master in 2131b405bfa9cc075ce54335b80dd41a1e85f552. What command did you run to get these failures? It should only be happening when running tests.
